wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,175 @@
+        self.sites = {
+            site.pk: site for site in Site.objects.all().prefetch_related("root_page")
+        }
+        self.sites_filter_enabled = True
+        if len(self.sites) == 1:
+            # If there is only one site, we don't need to show the site filter
+            self.sites_filter_enabled = False
+            del self.filters["site"]
+
+
+class PageTypesUsageReportView(ReportView):
+    template_name = "wagtailadmin/reports/page_types_usage.html"
+    title = _("Page types usage")
+    header_icon = "doc-empty-inverse"
+    filterset_class = PageTypesUsageReportFilterSet
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.user_model = get_user_model()
+        self.page_models = [model.__name__.lower() for model in get_page_models()]
+        self.i18n_enabled = getattr(settings, "WAGTAIL_I18N_ENABLED", False)
+
+    def user_id_to_python(self, user_id):
+        return self.user_model._meta.pk.to_python(user_id)
+
+    def add_last_edited_name_to_page_type(self, username_mapping, page_type):
+        if page_type.last_edited_page_owner_id:
+            # Giving the last_published_by annotation an explicit output_field type
+            # causes an issue with prefetch_workflow_states when the field is a
+            # ConvertedValueField. If the last user to publish the page has been
+            # deleted, we will render their user id in the template, so we call
+            # to_python on the value so that what's rendered matches the developer's
+            # expectation in the case of complex primary key types (e.g. UUIDField).
+            try:
+                user_id_value = self.user_id_to_python(
+                    page_type.last_edited_page_owner_id
+                )
+            except ValidationError:
+                user_id_value = page_type.last_edited_page_owner_id
+
+            last_edited_page_owner = username_mapping.get(
+                user_id_value, get_deleted_user_display_name(user_id=user_id_value)
+            )
+            page_type.last_edited_page_owner = last_edited_page_owner
+
+    def add_last_edited_page_to_page_type(self, pages_mapping, page_type):
+        if page_type.last_edited_page_id:
+            page_type.last_edited_page = pages_mapping.get(
+                page_type.last_edited_page_id
+            )
+
+    def decorate_paginated_queryset(self, page_types):
+        page_ids = set(page_types.values_list("last_edited_page_id", flat=True))
+        pages = Page.objects.filter(pk__in=page_ids).select_related("owner")
+
+        pages_mapping = {page.pk: page for page in pages}
+        username_mapping = {
+            page.owner.pk: page.owner.get_username() for page in pages if page.owner
+        }
+
+        for page_type in page_types:
+            self.add_last_edited_page_to_page_type(pages_mapping, page_type)
+            self.add_last_edited_name_to_page_type(username_mapping, page_type)
```

Comment:
If we dropped the separate username mapping, this could be as simple as:
```suggestion
        for item in page_types:
            item.last_edited_page = pages_mapping[item.last_edited_page_id] if item.last_updated_page_id else None
```

Reply:
I don't know if we can do that though. Besides mapping the user id to a user object, the mapping also handles the cases where the user was deleted (hence the `get_deleted_user_display_name`) -- we might still need a list comprehension for that

Reply:
That piece of code was "inspired" from the [aging pages report](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/views/reports/aging_pages.py#L68-L84) as well, but I think the reports should be consistent in how they display the user names...

Reply:
I think we can find a better solution here... How about adding a property to `Page` that returns the `owner.get_full_name()` when the user exists, `get_deleted_user_display_name(self.owner_id)` when it doesn't, or `""` when `self.owner_id is None`?

Reply:
Not displaying the owner anymore :)